src/common/transformations/tests/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -433,3 +433,64 @@ TEST_F(SharedTransformationTestsF, SharedShapeOfTestMixed) {
         model_ref = std::make_shared<Model>(NodeVector{concat}, ParameterVector{input});
     }
 }
+
+namespace {
+template <typename T, typename... NodeArgTypes>
+OutputVector createNodesInMemory(const std::vector<size_t>& node_order,
+                                 std::shared_ptr<T[]>& memory,
+                                 const std::string& node_name_prefix,
+                                 NodeArgTypes&&... node_args) {
+    OutputVector outputs;
+    memory.reset(new T[node_order.size()]);
```

Comment:
could you check that a destructor of each ShapeOf obj in this array will be called in this case?